### ISSUE TITLE
GP optimization doesn't work when defined as a separate step-based operation

### ISSUE THREAD

**Issue by jukiewiczm** [guildai/guildai#126]
Minimal example:
script
```
cat test.py 
import math

x = 10
y = 1

print(f"loss: {math.sin(x) - math.cos(y)}")
```

guild.yml
```
- model: test_model
  operations:
    evaluate:
      main: test
      flags:
        x:
          default: 94
    tune:
      steps:
        - run: evaluate 
            x="[0:100]" 
            --optimizer gp 
            --max-trials 80
            --minimize loss
```

This works just fine:
```
guild run evaluate x=[0:100] --optimizer gp --max-trials 80  
```

This doesn't really work
```
guild run tune
```

Because what happens is the following:
```
INFO: [guild] Random start for optimization (1 of 3)
INFO: [guild] Random start for optimization (2 of 3)
INFO: [guild] Random start for optimization (3 of 3)
INFO: [guild] Random start for optimization (missing previous trials)
INFO: [guild] Random start for optimization (missing previous trials)
INFO: [guild] Random start for optimization (missing previous trials)
...
```

It seems it's unable to identify the previous runs properly and fails.

It worked up to `0.7.0rc3`
It broke in `0.7.0rc4`

**Comment by gar1t** [comment]
Thanks for the report on this! This has been fixed in master and will be available in 0.7.0.rc10.

**Comment by gar1t** [comment]
The fix is now generally available in 0.7.0. If this is still an issue, please re-open and we'll sort it out. Thanks again for the report!
